### PR TITLE
Updates signaling of features and presences in the room

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: src/main/java/org/jitsi/jigasi/JvbConference.java [hunk 1]
```diff
@@ -1846,6 +1856,19 @@ public void onPasswordReceived(String pwd)
         joinConferenceRoom();
     }
 
+    /**
+     * Creates a feature xmpp extension, that can be added to features and used in presence.
+     * @param var the value to be added.
+     * @return the extension element.
+     */
+    private static ExtensionElement getFeature(String var)
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/JvbConference.java:1890]: nit: Statics on top, maybe name `createFeature`

File: src/main/java/org/jitsi/jigasi/JvbConference.java [hunk 2]
```diff
@@ -1201,20 +1212,9 @@ private void processChatRoomMemberLeft(ChatRoomMember member)
...
-                        DiscoverInfo i = ServiceDiscoveryManager.getInstanceFor(getConnection())
-                            .discoverInfo(JidCreate.entityFullFrom(jidString));
-
-                        if (!i.containsFeature(SIP_GATEWAY_FEATURE_NAME))
-                        {
-                            return true;
-                        }
-                    }
-                    catch(Exception e)
+                    if (!jigasiChatRoomMembers.contains(m.getName()))
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/JvbConference.java]: `return !jigasiChatRoomMembers.contains(m.getName())`?

I find all this hard to read, I'd do something like 
```
onlyJigasisInRoom = allMatch(m ->
  m.getName().equals(getResourceIdentifier().toString() 
  || m.getName().equals(gatewaySession.getFocusResourceAddr()) 
  || jigasiChatRoomMembers.contains(m.getName()));
 ```

Comment (bgrozev) [reply src/main/java/org/jitsi/jigasi/JvbConference.java]: There's also `ChatRoomMemberJabberImpl#getLastPresence()` that you can use to make a `isJigasi` utility an avoid saving jigasi names in a list.

Comment (damencho) [reply src/main/java/org/jitsi/jigasi/JvbConference.java]: There was something about it that prevents me from using it ... I need to double check

File: src/main/java/org/jitsi/jigasi/JvbConference.java [hunk 3]
```diff
@@ -1190,62 +1231,48 @@ private void processChatRoomMemberLeft(ChatRoomMember member)
             return;
         }
 
-        try
+        // if lobby is not enabled or single moderator mode is detected
+        // there is nothing to process
+        // but otherwise we will check whether there are
+        if (this.lobbyEnabled && !this.singleModeratorEnabled)
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/JvbConference.java:1237]: The doc isn't finished. ill check whether there are ...?

File: src/main/java/org/jitsi/jigasi/JvbConference.java (hunk 2 above)
Comment (damencho) [reply src/main/java/org/jitsi/jigasi/JvbConference.java]: I cannot use getLastPresence(), cause jitsi xmpp PP does not store the first presence ... only the updates after creating the member. I will leave it like this, as it needs more refactoring in jitsi-desktop.

**Review by bgrozev — APPROVED** [review]

**Pull request merged**